packages/react-component-library/src/primitives/FloatingBox/FloatingBox.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { forwardRef } from 'react'
+import classNames from 'classnames'
+
+import { FLOATING_BOX_SCHEME, FLOATING_BOX_ARROW_POSITION } from './constants'
+
+export interface FloatingBoxProps extends PositionType, ComponentWithClass {
+  width?: number
+  height?: number
+  top?: number
+  right?: number
+  bottom?: number
+  left?: number
+  scheme?: FLOATING_BOX_SCHEME.LIGHT | FLOATING_BOX_SCHEME.DARK
+  position?:
+    | FLOATING_BOX_ARROW_POSITION.LEFT_BOTTOM
+    | FLOATING_BOX_ARROW_POSITION.LEFT_TOP
+    | FLOATING_BOX_ARROW_POSITION.RIGHT_BOTTOM
+    | FLOATING_BOX_ARROW_POSITION.RIGHT_TOP
+    | FLOATING_BOX_ARROW_POSITION.TOP_LEFT
+    | FLOATING_BOX_ARROW_POSITION.TOP_RIGHT
+    | FLOATING_BOX_ARROW_POSITION.BOTTOM_LEFT
+    | FLOATING_BOX_ARROW_POSITION.BOTTOM_RIGHT
+  onMouseEnter?: (e: React.MouseEvent) => void
+  onMouseLeave?: (e: React.MouseEvent) => void
+  children?: React.ReactElement
+}
+
+export const FloatingBox = forwardRef(
```

Comment:
This could also be declared as `forwardRef<FloatingBoxProps, React.Ref<HTMLDivElement>`. Not sure what reads nicer 🤷‍♂

Reply:
Not quite sure how you mean it could be different - could you elaborate further?

Reply:
Apologies for not being clear:

```
export const FloatingBox = forwardRef<FloatingBoxProps, any>(({ ... }, ref) => {
  return (
    ...
  )
})